fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,180 @@
+                     plot_error_labels: bool=True,
+                     text_precision_digits: int=4,
+                     text_fontsize: int=8,
+                     text_color: str="black",
+                     text_ha: str="center",
+                     capsize=10,
+                     colormap="Pastel1",
+                     subplot_shape=None,
+                     figsize=None
+                     ):
+    """Visualization for metrics with statistical error bounds.
+
+    Plots a given metric and its given error (as described by the `error_mapping`)
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics and metric errors
+    and a `error_mapping` dict to interpret the columns (??) of the `MetricFrame`.
+    
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection fo disaggregated metric values, along with the metric errors.
+
+    plot_type : str
+        The type of plot to display. i.e. "bar", "line", etc
+
+    metrics : str or list of str
+        The name of the metrics to plot. Should match columns from the given `MetricFrame`
+
+    axs : matplotlib.axes._axes.Axes, optional
+        Custom Matplotlib axes to which this function can plot
+
+    show_plot : bool, optional
+        Whether to show the plot or not
+
+    plot_error_labels : bool, optional
+        Whether or not to plot numerical labels for the error bounds
+
+    text_precision_digits : int, optional
+        The number of digits of precision to show for error labels
+
+    text_font_size : int, optional
+        The font size to use for error labels
+
+    text_color : str, optional
+        The font color to use for error labels
+
+    text_ha : str, optional
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    matplotlib.axes._axes.Axes
+    """
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+    # TODO: Check if metrics and error_bars/conf_intervals same length 
+
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+    
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    error_bars = [error_bars] if isinstance(error_bars, str) else error_bars
+
+    df = metric_frame.by_group
+
+    # only plot metrics that aren't tuples
+    if metrics is None:
+        metrics = []
+        for metric in list(df):
+            if not isinstance(df[metric][0], tuple) and metric != "_index":
+                metrics.append(metric)
+
```

Comment:
Should there be a check after this that `metrics` isn't an empty list? It might be an expected case.... what happens then?

Reply:
It will plot an empty plot. Do we want to change to an error message?

Reply:
I'll defer to others on the best behaviour.

Reply:
Potentially stupid question: What is an empty plot?

Reply:
It's like an all-white blank plot with no points/bars/etc

Reply:
<img width="681" alt="image" src="https://user-images.githubusercontent.com/24417440/125912934-19f02a4b-d2e6-4493-92ab-b6a3d5b7ab76.png">

Reply:
What would be the point of this? 🤣

Reply:
An error message or at least a warning would probably make more sense here.